### PR TITLE
Adm 503 Display correct emoj for pipeline step in metrics page and export page

### PR DESCRIPTION
**Pull request by WSSsssss33**
## Summary
* Display correct emoj for pipeline step in metrics page and export page
...

## Before
image
image

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## After

image
image

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## Note

_Null_

### CODE REVIEW

File: frontend/src/components/Common/ReportForThreeColumns/index.tsx [hunk 1]
```diff
@@ -13,11 +16,27 @@ interface ReportForThreeColumnsProps {
 }
 
 export const ReportForThreeColumns = ({ title, fieldName, listName, data }: ReportForThreeColumnsProps) => {
+  const emojiRow = (row: ReportDataWithThreeColumns) => {
+    if (row.name.includes(':') && getEmojiUrls(row.name, emojis).length > 0) {
+      const prefix = row.name.split('/')[0]
```
Comment (ziwen-xiao) [inline frontend/src/components/Common/ReportForThreeColumns/index.tsx:21]: Personally I would prefer the destructuring approach here
`const [prefix, suffix] = row.name.split('/')`
as using index to get the array values may slightly reduce readability.

File: frontend/src/utils/util.ts [hunk 2]
```diff
@@ -19,3 +19,32 @@ export const downloadCSV = (filename: string, data: string) => {
...
+export interface BuildKiteEmoji {
+  name: string
+  url: string
+  aliases?: string[]
+}
+
+export const getEmojiUrls = (input: string, emojis: BuildKiteEmoji[]): string[] => {
+  const names = getEmojiNames(input)
+  return names.flatMap((name) => {
+    const emoji: BuildKiteEmoji = emojis.find((emoji) => emoji.name === name) as BuildKiteEmoji
```
Comment (ziwen-xiao) [inline frontend/src/utils/util.ts:32]: seems we don't need the cast statement here as we have specified the emojis as `BuildKiteEmoji []`

File: frontend/src/utils/util.ts [hunk 3]
```diff
@@ -19,3 +19,32 @@ export const downloadCSV = (filename: string, data: string) => {
...
+
+export const getEmojiNames = (input: string): string[] => {
+  const regex = /:([\w+-]+):/g
+  const matches = input.match(regex) || []
+  return matches.map((match) => match.replaceAll(':', ''))
+}
+
+export const removeExtraEmojiName = (input: string): string => {
+  const names = getEmojiNames(input)
+  let output = input
```
Comment (ziwen-xiao) [inline frontend/src/utils/util.ts:45]: wondering if it's necessary to introduce an extra `output` here, i guess we can simply use the input value?

File: frontend/src/components/Common/ReportForThreeColumns/index.tsx (hunk 1 above)
Comment (WSSsssss33) [reply frontend/src/components/Common/ReportForThreeColumns/index.tsx:21]: nice syntactic sugar!

File: frontend/src/utils/util.ts (hunk 2 above)
Comment (WSSsssss33) [reply frontend/src/utils/util.ts:32]: test not work if there is no-type-statement

Comment (ziwen-xiao) [reply frontend/src/utils/util.ts:32]: ok then let's keep it here

**Review by ziwen-xiao — APPROVED** [review]

**Review by TingLiMio — APPROVED** [review]

File: frontend/src/components/Common/ReportForThreeColumns/style.tsx [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { styled } from '@mui/material/styles'
+import { Avatar } from '@mui/material'
+
+export const StyledAvatar = styled(Avatar)({
+  width: '20px',
```
Comment (guzhongren) [inline frontend/src/components/Common/ReportForThreeColumns/style.tsx:5]: Should NOT use `px` in frontend, use `rem`

File: frontend/src/components/Metrics/MetricsStep/DeploymentFrequencySettings/SingleSelection/style.tsx [hunk 5]
```diff
@@ -1,8 +1,14 @@
...
-import { FormControl } from '@mui/material'
+import { Avatar, FormControl } from '@mui/material'
 
 export const FormControlWrapper = styled(FormControl)({
   margin: '0 0 2rem 2.5%',
   width: '95%',
   height: '2.5rem',
 })
+
+export const StyledAvatar = styled(Avatar)({
```
Comment (guzhongren) [inline frontend/src/components/Metrics/MetricsStep/DeploymentFrequencySettings/SingleSelection/style.tsx:10]: duplicated code, ref to frontend/src/components/Common/ReportForThreeColumns/style.tsx

File: frontend/src/assets/emojis.json
Comment (guzhongren) [inline frontend/src/assets/emojis.json:1]: Store this emoji list here is not a good solution. Maybe we can use github cdn to keep the `source of truth` for this data. 

- Fetch emoji list when pages come to metric
- Store it in localStorage or redux
- user it in the avatar

cdn can ref to our `data-hosting` project, 
e.g. https://cdn.jsdelivr.net/gh/buildkite/emojis@main/img-buildkite-64.json

**Review by guzhongren — COMMENTED** [review]
Some code smells and 👇

Store this emoji list here is not a good solution. Maybe we can use github cdn to keep the `source of truth` for this data. 

- Fetch emoji list when pages come to metric
- Store it in localStorage or redux
- user it in the avatar

cdn can ref to our `data-hosting` project, 
e.g. https://cdn.jsdelivr.net/gh/buildkite/emojis@main/img-buildkite-64.json

**Pull request merged**